cli/src/pcluster/models/common.py

```diff
@@ -239,3 +251,330 @@ def dump_json(self):
+        :return: 16 chars string e.g. 2238a84ac8a74529
+        """
+        return hashlib.sha256((account_id + region).encode()).hexdigest()[0:16]
+
+    def check_bucket_exists(self):
+        """Check bucket existence by bucket name."""
+        AWSApi.instance().s3.head_bucket(bucket_name=self.name)
+
+    def create_bucket(self):
+        """
+        Create a new S3 bucket.
+
+        :param bucket_name: name of the S3 bucket to create
+        :param region: aws region
+        """
+        if self.region != "us-east-1":
+            AWSApi.instance().s3.create_bucket(bucket_name=self.name, configuration={"LocationConstraint": self.region})
+        else:
+            AWSApi.instance().s3.create_bucket(bucket_name=self.name)
+
+    def configure_s3_bucket(self):
+        """Configure s3 bucket to satisfy pcluster setting."""
+        AWSApi.instance().s3.put_bucket_versioning(bucket_name=self.name, configuration={"Status": "Enabled"})
+        AWSApi.instance().s3.put_bucket_encryption(
+            bucket_name=self.name,
+            configuration={"Rules": [{"ApplyServerSideEncryptionByDefault": {"SSEAlgorithm": "AES256"}}]},
+        )
+        deny_http_policy = (
+            '{{"Id":"DenyHTTP","Version":"2012-10-17","Statement":[{{"Sid":"AllowSSLRequestsOnly","Action":"s3:*",'
+            '"Effect":"Deny","Resource":["arn:{partition}:s3:::{bucket_name}","arn:{partition}:s3:::{bucket_name}/*"],'
+            '"Condition":{{"Bool":{{"aws:SecureTransport":"false"}}}},"Principal":"*"}}]}}'
+        ).format(bucket_name=self.name, partition=self.partition)
+        AWSApi.instance().s3.put_bucket_policy(bucket_name=self.name, policy=deny_http_policy)
+
+    # --------------------------------------- S3 objects utils --------------------------------------- #
+
+    def get_object_key(self, object_type, object_name):
+        """Get object key of an artifact."""
+        return "/".join([self.artifact_directory, object_type, object_name])
+
+    def delete_s3_artifacts(self):
+        """Cleanup S3 bucket artifact directory."""
+        LOGGER.debug(
+            "Cleaning up S3 resources bucket_name=%s, service_name=%s, remove_artifact=%s",
+            self.name,
+            self._service_name,
+            self._cleanup_on_deletion,
+        )
+        if self.artifact_directory and self._cleanup_on_deletion:
+            try:
+                LOGGER.info("Deleting artifacts under %s/%s", self.name, self.artifact_directory)
+                AWSApi.instance().s3_resource.delete_object(
+                    bucket_name=self.name, prefix="%s/" % self.artifact_directory
+                )
+                AWSApi.instance().s3_resource.delete_object_versions(
+                    bucket_name=self.name, prefix="%s/" % self.artifact_directory
+                )
+            except AWSClientError as e:
+                LOGGER.warning(
+                    "Failed to delete S3 artifact under %s/%s with error %s. Please delete them manually.",
```

Comment:
`%s/%s` includes bucket name and artifact directory. But I'd avoid raising error here, because it's not neccessary to raise an error when artifacts directory deletion fails.